datashader/bokeh_ext.py

```diff
@@ -228,7 +265,14 @@ def _repr_html_(self):
         for m in self.p.references():
             m._document = None
         self.doc.add_root(self.p)
-        return notebook_div(self.p, self.ref)
+        if bokeh_version > '0.12.9':
+            js, div, _ = notebook_content(self.p, self.ref)
+            html = NOTEBOOK_DIV.format(plot_script=js, plot_div=div)
+            div = encode_utf8(html)
+            self.doc.hold()
+        else:
+            div = notebook_div(self.p, self.ref)
+        return div
```

Comment:
Sigh!  Bokeh 0.12.10 sure seems to need more verbose code.  Again, making the new bit into a function with a name that explains what the heck it achieves would make this class be so much more readable...